base/loading.jl

```diff
@@ -2819,11 +2819,20 @@ global mkpidlock_hook
 global trymkpidlock_hook
 global parse_pidfile_hook
 
-# allows processes to wait if another process is precompiling a given source already
-function maybe_cachefile_lock(f, pkg::PkgId, srcpath::String)
+# The preferences hash is only known after precompilation so just assume no preferences
+# meaning that if all other conditions are equal, the same package cannot be precompiled
+# with different preferences at the same time.
+compilecache_pidfile_path(pkg::PkgId) = compilecache_path(pkg, UInt64(0)) * ".pidfile"
+
+# Allows processes to wait if another process is precompiling a given source already.
+# The lock file is deleted and precompilation will proceed after `stale_age` seconds if
+#  - the locking process no longer exists
+#  - the lock is held by another host, since processes cannot be checked remotely
+# or after `stale_age * 25` seconds if it does still exist.
+function maybe_cachefile_lock(f, pkg::PkgId, srcpath::String; stale_age=60)
```

Comment:
This is probably too short. It is supposed to be a lower bound, not an upper bound. Something around 300 is probably okay (5 minutes) though, or perhaps even 600.

Reply:
Changed in https://github.com/JuliaLang/julia/pull/50254